src/test/compile-fail/return-from-diverging.rs

```diff
@@ -8,9 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-// pretty-expanded FIXME #23616
+fn fail() -> ! {
+    return "wow"; //~ ERROR mismatched types
```

Comment:
Hmm. I wonder if we want a customized error message in this scenario -- seems likely (and probably applies to all uses of `!` equally). Maybe another good item to follow for follow-up.

cc @jonathandturner -- maybe we can add such a thing to some list :)